### ISSUE TITLE
Failed to run `docker run hello-world` on RPi 2

### ISSUE THREAD

**Issue by poliveira89** [docker/docker#29710]
**Description**

Docker Engine installed on Raspbian Jessie for a Raspberry Pi 2.

```
$ uname -srmo
Linux 4.4.38-v7+ armv7l GNU/Linux
```

**Steps to reproduce the issue:**
1. Installed alongside this [tutorial](https://docs.docker.com/engine/installation/linux/raspbian/)
2. Request image `docker pull hello-world`
3. Execute the command `docker run hello-world`

**Describe the results you received:**

```
$  docker run hello-world
panic: standard_init_linux.go:175: exec user process caused "exec format error" [recovered]
	panic: standard_init_linux.go:175: exec user process caused "exec format error"

goroutine 1 [running, locked to thread]:
panic(0x3360b0, 0x56e57300)
	/usr/local/go/src/runtime/panic.go:481 +0x330
github.com/urfave/cli.HandleAction.func1(0x56ea3958)
	/go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/app.go:478 +0x328
panic(0x3360b0, 0x56e57300)
	/usr/local/go/src/runtime/panic.go:443 +0x448
github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization.func1(0x56ea35e0, 0x56dda568, 0x56ea3668)
	/go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:259 +0x13c
github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization(0x56e54b10, 0x46430620, 0x56e57300)
	/go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:277 +0x498
main.glob.func8(0x56e701e0, 0x0, 0x0)
	/go/src/github.com/opencontainers/runc/main_unix.go:26 +0x58
reflect.Value.call(0x2eb600, 0x3f2948, 0x13, 0x3648b8, 0x4, 0x56ea3918, 0x1, 0x1, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:435 +0xeb4
reflect.Value.Call(0x2eb600, 0x3f2948, 0x13, 0x56ea3918, 0x1, 0x1, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:303 +0x84
github.com/urfave/cli.HandleAction(0x2eb600, 0x3f2948, 0x56e701e0, 0x0, 0x0)
	/go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/app.go:487 +0x230
github.com/urfave/cli.Command.Run(0x366cd8, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3d0d50, 0x51, 0x0, ...)
	/go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/command.go:191 +0xcd0
github.com/urfave/cli.(*App).Run(0x56e40240, 0x56e22050, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/app.go:240 +0x84c
main.main()
	/go/src/github.com/opencontainers/runc/main.go:137 +0xbb4
```

**Describe the results you expected:**

Container started sucessfully

**Additional information you deem important (e.g. issue happens only occasionally):**

The log after the command executed:

```
$ journalctl -u docker.service
(...)
Dec 26 16:00:30 mars dockerd[565]: time="2016-12-26T16:00:30.839362187Z" level=info msg="No non-localhost DNS nameservers are left in resolv.conf. Using default external servers : [nameserver 8.8.8.8 nameserver 8.8.4.4]"
Dec 26 16:00:30 mars dockerd[565]: time="2016-12-26T16:00:30.839725518Z" level=info msg="IPv6 enabled; Adding default IPv6 external servers : [nameserver 2001:4860:4860::8888 nameserver 2001:4860:4860::8844]"
Dec 26 16:00:31 mars dockerd[565]: time="2016-12-26T16:00:31.620129369Z" level=warning msg="failed to close stdin: rpc error: code = 2 desc = containerd: container not found"
```

**Output of `docker version`:**

```
$ docker version
Client:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 03:23:36 2016
 OS/Arch:      linux/arm

Server:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 03:23:36 2016
 OS/Arch:      linux/arm
```

**Output of `docker info`:**

```
$ docker info
Containers: 5
 Running: 0
 Paused: 0
 Stopped: 5
Images: 1
Server Version: 1.12.5
Storage Driver: devicemapper
 Pool Name: docker-179:2-258054-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 310.4 MB
 Data Space Total: 107.4 GB
 Data Space Available: 29.83 GB
 Metadata Space Used: 970.8 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.90 (2014-09-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host null bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options:
Kernel Version: 4.4.38-v7+
Operating System: Raspbian GNU/Linux 8 (jessie)
OSType: linux
Architecture: armv7l
CPUs: 4
Total Memory: 925.5 MiB
Name: mars
ID: U7AC:EMNY:DWXF:OUEN:YAWZ:VBDT:6RRI:PXML:AVG7:C6DE:LN67:B23C
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
WARNING: No cpuset support
Insecure Registries:
 127.0.0.0/8
```

**Comment by vdemeester** [comment]
@poliveira89 The `helloworld` image is for `amd64` I guess, so the binary inside it have not the right architecture, hence the error : `exec format error`.

**Comment by poliveira89** [comment]
@vdemeester I did not knew about that, but it's on the Raspbian tutorial. That being said, I thought that if the `hello-world` was not working should be related to engine.

But you seem to be right, because I pulled this image `container4armhf/armhf-alpine` and then run:
```
$ docker run container4armhf/armhf-alpine /bin/echo 'Hello world'
Hello world
```

And it worked! So, I think this after is a non-issue for that, I will close it. Thanks.
